src/ConnectionPool.jl

```diff
@@ -325,22 +325,40 @@ function purge(c::Connection)
     @ensure bytesavailable(c) == 0
 end
 
+const TCP_POOL = Pool(Connection{Sockets.TCPSocket})
+const MbedTLS_SSL_POOL = Pool(Connection{MbedTLS.SSLContext})
+const OpenSSL_SSL_POOL = Pool(Connection{OpenSSL.SSLStream})
 """
-    closeall()
+    POOLS
 
-Close all connections in`pool`.
+A dict of global connection pools keeping track of active connections, split by their IO type.
 """
-function closeall()
-    ConnectionPools.reset!(POOL)
-    return
+const POOLS = Dict{DataType,Pool}(
+    Sockets.TCPSocket => TCP_POOL,
+    MbedTLS.SSLContext => MbedTLS_SSL_POOL,
+    OpenSSL.SSLStream => OpenSSL_SSL_POOL,
+)
+getpool(::Type{Sockets.TCPSocket}) = TCP_POOL
+getpool(::Type{MbedTLS.SSLContext}) = MbedTLS_SSL_POOL
+getpool(::Type{OpenSSL.SSLStream}) = OpenSSL_SSL_POOL
+# Fallback for custom connection io types
+# to opt out from locking, define your own `Pool` and add a `getpool` method for your IO type
+const POOLS_LOCK = Threads.ReentrantLock()
+function getpool(::Type{T}) where {T}
+    lock(POOLS_LOCK)
+    try
+        get!(POOLS, T, Pool(Connection{T}))::Pool{Connection{T}}
```

Comment:
we probably don't want to actuall call `Pool` if we don't need to, as that's going to allocate a Dict 

```suggestion
        get!(() -> Pool(Connection{T}), POOLS, T)::Pool{Connection{T}}
```

Reply:
Ooof, good catch